ext/src/nassau.rs

```diff
@@ -130,6 +135,67 @@ impl MilnorSubalgebra {
     pub fn iter_signatures(&self, degree: i32) -> impl Iterator<Item = Vec<PPartEntry>> + '_ {
         SignatureIterator::new(self, degree)
     }
+
+    fn top_degree(&self) -> i32 {
+        self.profile
+            .iter()
+            .map(|&entry| 2i32.pow(entry as u32) - 1)
```

Comment:
I personally prefer using bitshift operators, which is what most of the codebase does. I don't have very strong opinions though.